.github/workflows/test.yaml

```diff
@@ -28,4 +28,88 @@ jobs:
       - name: Run Go Tests
         if: env.GIT_DIFF
         run: |
-          go list -f "{{ if (or .TestGoFiles .XTestGoFiles) }}{{ .ImportPath }}{{ end }}" ./... | xargs -t go test -v 
+          go list -f "{{ if (or .TestGoFiles .XTestGoFiles) }}{{ .ImportPath }}{{ end }}" ./... | xargs -t go test -v
+
+  e2e:
```

Comment:
my inclination, and this is somewhat counter to the example in the cloud repo, is to keep different suites/classes of tests in different workflow files just to make things easier to find/trace, particularly if there aren't dependencies between them.